### PR TITLE
Adding the ability to granularly display disk space ranges

### PR DESCRIPTION
**Pull request by stratus-ss**
This is a big PR that deals a lot with breaking apart CLIOutput into smaller, resuable chunks. Additional tests have been added to provide more coverage CLIOutput

### CODE REVIEW

File: vminfo_parser/analyzer.py [hunk 1]
```diff
@@ -264,44 +362,62 @@ def handle_disk_space(
...
         self.cli_output.print_formatted_disk_space(
-            sorted_range_counts_by_environment,
-            environment_filter,
-            env_keywords,
+            col_widths,
+            formatted_df_str,
             os_filter=os_filter,
+            index_heading_justification=column_padding["index_heading_justification"],
+            other_headings_justification=column_padding["other_headings_justification"],
         )
```
Comment (csfreak) [inline vminfo_parser/analyzer.py:421]: Why is format_rows being called by analyzer, and brought back into analyzer if its only sent back to cli_output?

File: vminfo_parser/clioutput.py [hunk 2]
```diff
@@ -86,30 +86,30 @@ def format_series_output(self: t.Self, counts: pd.Series) -> None:
...
-        if environment_filter == "all":
-            formatted_rows.append("Disk Space Range".ljust(20) + "Count".ljust(col_widths["Count"]))
-            justification = 19
-        else:
-            justification = 15
+        dataFrame: pd.DataFrame,
+        formatted_rows: list,
+        justification: int,
+        col_widths: dict,
+    ):
```
Comment (csfreak) [inline vminfo_parser/clioutput.py:95]: Missing return annotation

File: vminfo_parser/config.py [hunk 3]
```diff
@@ -65,6 +65,12 @@ def _get_parser() -> argparse.ArgumentParser:
         default=False,
         help="Show disk space by OS",
     )
+    parser.add_argument(
+        "--disk-space-by-granular-os",
+        action="store_true",
+        default=False,
+        help="When getting disk space by os, this breaks those results more granularly",
+    )
```
Comment (csfreak) [inline vminfo_parser/config.py:73]: If we are going to add config flags,  we should add E2E test cases for them,  especially until we have unit tests for main and analyzer written

**Review by csfreak — CHANGES_REQUESTED** [review]
While I think we need to refactor the tests to use dictionaries in `tests/consts.py` I dont think test formatting is a reason to delay the PR.    Other issues can be discussed, or fixed on PR to main if you want.

File: vminfo_parser/analyzer.py (hunk 1 above)
Comment (stratus-ss) [reply vminfo_parser/analyzer.py:421]: I was trying to break apart  the print_formatted_disk_space. Initially it had all of this logic inside of it, including set_column_width()

Flake8 was tripping over the complexity with both of these tasks in the print_formatted_disk_space() so I broken them up.

File: vminfo_parser/clioutput.py (hunk 2 above)
Comment (stratus-ss) [reply vminfo_parser/clioutput.py]: fixed

File: vminfo_parser/analyzer.py (hunk 1 above)
Comment (csfreak) [reply vminfo_parser/analyzer.py:421]: I agree with splitting them up,  but I think that the logic can and should be called from within clioutput, unless the output is needed in analyzer for something else

Comment (csfreak) [reply vminfo_parser/analyzer.py:421]: If format_rows is only useful to cli_output,  then analyzer shouldn't know or care about it,  ultimately, the goal is analyzer doesn't care what the output is,  its just crunching data.

**Review by csfreak — APPROVED** [review]

**Pull request merged**